freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/keytab/model/UserKeytabRequest.java

```diff
@@ -0,0 +1,63 @@
+package com.sequenceiq.freeipa.api.v1.keytab.model;
+
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.sequenceiq.freeipa.api.v1.keytab.doc.KeytabModelDescription;
+import com.sequenceiq.service.api.doc.ModelDescriptions;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel("UserKeytabV1Request")
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class UserKeytabRequest {
+
```

Comment:
Is there a reason the full kerberos principal is not constructable from this request (where is the REALM)?

Although there is only 1 realm right now and it could be auto detected now. I am not sure if other code makes this assumption or not. My thought is that in the future there might be a data migration from env1 to env2. In such a scenario there could be 2 realms involved. This if forward thinking since we don't have this requirement now, but I want to consider API compatibility with potential future requirements.

Reply:
I'm planning to get the realm from the FreeIpa entity in that cluster. That information is available in the FreeIpaManagement service. That is why I was not planning to it take as an input from the consumer. 

What are you suggesting here?

Reply:
sounds good, I just wanted to make sure we considered it.